### DISCUSSION TITLE
[@types/node] stream/consumers does not accept web streams

### DISCUSSION THREAD

**Discussion by pimterry** [DefinitelyTyped/DefinitelyTyped#65755] [Issues with a @types package] [answered]
Each of the stream methods in https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/stream/consumers.d.ts do not accept a web stream (either from node's stream/web module, or using the native lib.dom types). The 'ReadableStream' & `Readable` types referenced are both node.js native stream types.

Meanwhile, [the consumer docs](https://nodejs.org/api/webstreams.html#utility-consumers) link ReadableStream to the web streams definition, in addition to the Node native stream type (referenced as `stream.Readable`) and actually define consumers within the web streams documentation itself.

The various different very similarly named types are awkward and a bit confusing, but web streams should be accepted here.

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment, marked as answer]
Yes, looks like it should be like
```diff
-    import { Readable } from 'node:stream';
-    function buffer(stream: NodeJS.ReadableStream | Readable | AsyncIterator<any>): Promise<Buffer>;
...
+    import { ReadableStream } from 'node:stream/web';
+    function buffer(stream: NodeJS.ReadableStream | ReadableStream | AsyncIterator<any>): Promise<Buffer>;
...
```

PR is welcome!
